docs/pages/documentation/11-examples/example-twitter-streaming-public-tweets.md

```diff
@@ -0,0 +1,580 @@
+  Entity userEntity = userEntityType.addEntity();
+  Resource userResource = userResourceType.putResource(user);
+  return userEntity.resource(userResource);
+}
+```
+
+And finally, write a function for inserting a user only if it's not yet there in the knowledge graph.
+
+```java
+public static Entity insertUserIfNotExist(GraknGraph graknGraph, String screenName) {
+  QueryBuilder qb = graknGraph.graql();
+  return findUser(qb, screenName).orElse(insertUser(graknGraph, screenName));
+}
+```
+
+### Relating The Tweet To The User
+
+We're almost there with a complete tweet insertion functionality! There's only one thing left to do which is to relate the `tweet` entity with the `user` entity. Preserving this connection is crucial, after all.
+
+The following function will create a relation between the user and tweet that we specify.
+
+```java
+public static Relation insertUserTweetRelation(GraknGraph graknGraph, Entity user, Entity tweet) {
+  RelationType userTweetRelationType = graknGraph.getRelationType("user-tweet-relation");
+  RoleType postsType = graknGraph.getRoleType("posts");
+  RoleType postedByType = graknGraph.getRoleType("posted_by");
+
+  Relation userTweetRelation = userTweetRelationType.addRelation()
+      .addRolePlayer(postsType, user)
+      .addRolePlayer(postedByType, tweet);
+
+  return userTweetRelation;
+}
+```
+
+### Tweet Insertion Wrap Up
+
+Finally, let's wrap up by defining a function of which the sole responsibility is to execute all of the methods we have defined above.
+
+```java
+public static Relation insertUserTweet(GraknGraph graknGraph, String screenName, String tweet) {
+  Entity tweetEntity = insertTweet(graknGraph, tweet);
+  Entity userEntity = insertUserIfNotExist(graknGraph, screenName);
+  return insertUserTweetRelation(graknGraph, userEntity, tweetEntity);
+}
+```
+
+We're done with tweet insertion functionality! Next step: querying the stored data. Before we proceed, let's add the method we've just defined to the main method as shown below.
+
+```java
+public static void main(String[] args) {
+  try (GraknSession session = Grakn.session(graphImplementation, keyspace)) {
+    withGraknGraph(session, graknGraph -> initTweetOntology(graknGraph)); // initialize ontology
+
+    listenToTwitterStreamAsync(consumerKey, consumerSecret, accessToken, accessTokenSecret, (screenName, tweet) -> {
+      withGraknGraph(session, graknGraph -> insertUserTweet(graknGraph, screenName, tweet)); // insert tweet
+      // TODO: perform some meaningful queries on the inserted data
+    });
+  }
+
+```
+
+## Crafting Simple Queries Using GraQL
+
+We will perform a query which will count the number of tweets a user has posted since the program started. It can be achieved it by utilizing the aggregate query feature. Graql has been chosen over the graph API for this task because it is declarative and therefore much easier to use for complex queries than the graph API.
```

Comment:
[OPTIONAL]

> much easier to use for complex queries than the graph API.

`much easier to use for complex queries.` - Shorter and easier to read in my opinion